### PR TITLE
feat(social): create social features

### PR DESCRIPTION
**Pull request by hmartiins**
### **Proposed changes**

create social features

<br>

### CODE REVIEW

File: src/modules/social/services/photoPost/DeletePhotoPostService.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  constructor(
+    @inject('PhotoPostRepository')
+    private photoPostRepository: IPhotoPostRepository,
+  ) {}
+
+  public async execute({ photo_post_id }: IRequest): Promise<void> {
+    const photoPost = await this.photoPostRepository.findById(photo_post_id);
+
+    if (!photoPost) {
+      throw new AppError('Tourist Spot Photo not found.', 404);
```
Comment (devLeSilverio) [inline src/modules/social/services/photoPost/DeletePhotoPostService.ts]: This error message should be:
Post Photo not found.

File: src/modules/social/services/photoPost/DeletePhotoPostService.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    deleteImage('tourist-spot-images', photoPost.url);
```
Comment (devLeSilverio) [inline src/modules/social/services/photoPost/DeletePhotoPostService.ts]: This should be called:
'post-images'

File: src/modules/social/services/photoPost/ShowPhotoPostService.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import PhotoPost from '@modules/social/infra/typeorm/entities/PhotoPost';
+import IPhotoPostRepository from '../../repositories/IPhotoPostRepository';
+
+interface IRequest {
+  post_id: string;
+}
+
+@injectable()
+export default class DeletePhotoPostService {
```
Comment (devLeSilverio) [inline src/modules/social/services/photoPost/ShowPhotoPostService.ts]: This should be called:
ShowPhotoPostService

File: src/modules/social/services/post/UpdatePostService.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import IPostRepository from '../../repositories/IPostRepository';
+
+interface IRequest {
+  post_id: string;
+  user_id: string;
+  content: string;
+}
+
+@injectable()
+export default class UpdateInterestPointService {
```
Comment (devLeSilverio) [inline src/modules/social/services/post/UpdatePostService.ts]: This should be called:
UpdatePostService

**Review by devLeSilverio — CHANGES_REQUESTED** [review]
Everything seems to make a lot of sense, I just found syntax errors, other than that LGTM!

**Comment by hmartiins** [comment]
@devLeSilverio I just corrected what you asked for, I don't even know how I missed it lol

**Comment by devLeSilverio** [comment]
Normal, LGTML!!

**Pull request merged**